share/smack/top.py

```diff
@@ -609,14 +588,16 @@ def annotate_bpl(args):
         f.write(bpl)
 
 
-def property_selection(args):
-    selected_props = []
-    if args.only_check_valid_deref:
-        selected_props.append('valid_deref')
-    elif args.only_check_valid_free:
-        selected_props.append('valid_free')
-    elif args.only_check_memleak:
-        selected_props.append('valid_memtrack')
+def memsafety_subproperty_selection(args):
+    selected_props = {}
+    if 'memory-safety' in args.check:
```

Comment:
Actually, we don't even need this statement 😂

Reply:
I think we do since you can have both `memory-safety` and `valid-deref` selected. That will lead to a wrong thing being checked if we remove this statement. No?

Reply:
Ah, yes.